src/db_common.rs

```diff
@@ -22,6 +23,19 @@ impl DbInner {
         Ok(())
     }
 
+    pub(crate) fn replay_memtable(
+        &self,
+        replayed_memtable: ReplayedMemtable,
+    ) -> Result<(), SlateDBError> {
+        let mut guard = self.state.write();
+        let last_wal_id = replayed_memtable.last_wal_id;
+        guard.set_next_wal_id(last_wal_id + 1);
+        guard.update_last_seq(replayed_memtable.last_seq);
+        self.mono_clock.set_last_tick(replayed_memtable.last_tick)?;
+        guard.replace_memtable(replayed_memtable.table)?;
+        self.maybe_freeze_memtable(&mut guard, last_wal_id)
```

Comment:
If this doesn't actually freeze the memtable (which should never happen because we ask the wal iterator to buffer up tables of max l0 size), then the next call to `replace_memtable` will panic. Should we have a version of this like `freeze_memtable` that freezes without checking anything?

Reply:
Good idea. There was an alternative definition of `freeze_memtable`, so I had to refactor a little bit to make the APIs more consistent. The current existing `freeze_memtable` was used in the context of flushing, so I moved its logic into `flush_memtables` and created a separate `flush_immutable_memtables`. Let me know what you think.